### ISSUE TITLE
[Bug][Tune] Loading `ExperimentAnalysis` requires `Trainable` to be registered.

### ISSUE THREAD

**Issue by matthewdeng** [ray-project/ray#21212]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Tune

### What happened + What you expected to happen

When initializing an `ExperimentAnalysis` from a path (as described in [Analysis (tune.analysis)](https://docs.ray.io/en/releases-1.9.0/tune/api_docs/analysis.html#analysis-tune-analysis)), an error is raised if the `Trainable` has not been registered (with `register_trainable`).

As a user:
1. It is not clear why the `Trainable` needs to be registered to set up the `ExperimentAnalysis`. From the implementation, the dependency is defined through `ExperimentAnalysis` -> `Trial` -> `Trainable`, but I can't tell if there's any functionality of the `ExperimentAnalysis` that actually requires the `Trainable`.
2. The expectation to call `register_trainable` before initializing an `ExperimentAnalysis` is not well defined or documented. For a user-defined Trainable, the registration usually happens implicitly within the Tune job, so it is not obvious how to then explicitly register it in order to load an `ExperimentAnalysis`.

Preferably, an `ExperimentAnalysis` can be loaded without explicitly registering the Trainables beforehand. Alternatively, the expected workflow should be well documented (how to register the Trainable, what name to use, etc.).

### Versions / Dependencies

Ray 1.9

### Reproduction script

**Script 1:** 
1. Run a Tune experiment. 
2. Successfully load the `ExperimentAnalysis` (because the Trainable is registered).
```python
from ray import tune
from ray.tune import ExperimentAnalysis

def training_function(config, checkpoint_dir=None):
   tune.report(alpha=config["alpha"])

name = "example-experiment"

analysis = tune.run(
   training_function,
   name=name,
   config={"alpha": tune.grid_search([1, 2, 3])})

analysis = ExperimentAnalysis(f"~/ray_results/{name}")
print(analysis.get_best_trial(metric="alpha",mode="max"))
```

**Script 2:**
Unsuccessfully load the `ExperimentAnalysis` from the path created in `Script 1`. Both error messages are caused by the logic in [validate_trainable](https://github.com/ray-project/ray/blob/8d89e2d54638edfba8586a207bc49b35952673e1/python/ray/tune/registry.py#L38-L44).
```python
from ray.tune import ExperimentAnalysis

name = "example-experiment"
analysis = ExperimentAnalysis(f"~/ray_results/{name}")

print(analysis.get_best_trial(metric="alpha",mode="max"))
```
**Scenario 1:** Trainable is not registered & RLlib _is not_ installed.
```
Could not load trials from experiment checkpoint. This means your experiment checkpoint is likely faulty or incomplete, and you won't have access to all analysis methods. Observed error: No module named 'gym'
Could not find best trial. Did you pass the correct `metric` parameter?
None
```
**Scenario 2:** Trainable is not registered & RLlib _is_ installed.
```
Could not load trials from experiment checkpoint. This means your experiment checkpoint is likely faulty or incomplete, and you won't have access to all analysis methods. Observed error: Unknown trainable: training_function
Could not find best trial. Did you pass the correct `metric` parameter?
None
```

**Script 2:**
Successfully load the `ExperimentAnalysis` from the path created in `Script 1` by first registering the Trainable.
```python
from ray.tune import ExperimentAnalysis
from ray.tune import register_trainable

def training_function(config, checkpoint_dir=None):
   tune.report(alpha=config["alpha"])

register_trainable("training_function", training_function)

name = "example-experiment"
analysis = ExperimentAnalysis(f"~/ray_results/{name}")

print(analysis.get_best_trial(metric="alpha",mode="max"))
```

### Anything else

This issue originates from https://discuss.ray.io/t/load-prior-tune-run-results-from-disk/4495.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by scho97** [comment]
Hi, Matthew. Thank you for opening this issue. I am having the same error, and I was wondering if I should use the way you suggested in `Script 2` for now. Is there any update on this?

**Comment by matthewdeng** [comment]
Ah @scho97 I unfortunately don't have any updates on this at the time. I think you can use Solution 2 as a workaround in the meantime.

@yard1 would you be able to take a look at this? Is this a byproduct of merging `Analysis` with `ExperimentAnalysis`?

**Comment by Yard1** [comment]
No, it's been a thing for a while @matthewdeng. We'll probably have to end up creating something like a `ReadOnlyTrial` that would not have an associated trainable if one is not found. I will be taking a look at this next sprint.

**Comment by parshad1710** [comment]
Hi, I am still having same exact issue to load ExperimentAnalysis even after registering the trainable. I am not sure what is the workaround. I have latest v1.10.

`Could not load trials from experiment checkpoint. This means your experiment checkpoint is likely faulty or incomplete, and you won't have access to all analysis methods. Observed error: No module named 'gym'
`
I donot have any dependencies on the 'gym' module. Any other suggested workarounds?

Thanks!

**Comment by Yard1** [comment]
Can you install `gym`?

**Comment by parshad1710** [comment]
I installed gym and now the error turned to:

`Could not load trials from experiment checkpoint. This means your experiment checkpoint is likely faulty or incomplete, and you won't have access to all analysis methods. Observed error: argument should be bytes, buffer or ASCII string, not 'PlacementGroupFactory'`

Here's the code snippet:

```
import tune

analysis = tune.ExperimentAnalysis("path/to/experiment/trials/folder")
```

**Comment by Yard1** [comment]
@parshad1710 It looks like the fix hasn't been merged into Ray 1.10. You can [install the latest nightly version](https://docs.ray.io/en/latest/installation.html#daily-releases-nightlies) which will include it, or wait for Ray 1.11.
